### DISCUSSION TITLE
Starter creek blog - Hero images not showing on production

### DISCUSSION THREAD

**Discussion by 4722794** [statamic/cms#6843] [Troubleshooting]
Hi!
I tried deploying a basic statamic blog using digital ocean using the tutorial here: 

https://statamic.dev/deploying/digitalocean-app-platform

Everything is swell, but for some reason the hero images are not showing on the deployed version.

Link here: https://blog.sportshaala.com/

Could anyone point me to a setting or tutorial to resolve this?

### Local deployment
<img width="1135" alt="Screenshot 2022-10-07 at 3 40 34 PM" src="https://user-images.githubusercontent.com/51963940/194530586-19baa63d-a638-4f46-aca1-0e24a4e8a65a.png">

### Production deployment
<img width="567" alt="Screenshot 2022-10-07 at 3 40 41 PM" src="https://user-images.githubusercontent.com/51963940/194530723-9812f599-c692-40a0-81ad-7495109a42be.png">

**Comment by duncanmcclean** [comment]
Do you get any console errors?

**Comment by jackmcdade** [comment]
My guess is that your server doesn’t have GD r ImageMagick installed or enabled.  On Oct 7, 2022, at 6:17 AM, Duncan McClean ***@***.***> wrote:﻿
Do you get any console errors?

**Comment by 4722794** [comment]
Yes! 
@jackmcdade 
This is the error I'm getting
<img width="521" alt="Screen Shot 2022-10-08 at 6 28 27 PM" src="https://user-images.githubusercontent.com/51963940/194708800-7c982811-9da2-434a-ab67-475c0e8b8610.png">
The server I am using is Digital Ocean app. (followed the tutorial on the website)
It's hosted here: blog.sportshaala.com

**Comment by jackmcdade** [comment]
500 errors are server side, do you have any server error logs that might share the actual message?On Oct 8, 2022, at 9:00 AM, 4722794 ***@***.***> wrote:﻿
Yes!
@jackmcdade
This is the error I'm getting

The server I am using is Digital Ocean app. (followed the tutorial on the website)
It's hosted here: blog.sportshaala.com

**Comment by 4722794** [comment]
Hi @jackmcdade,
Here are the server logs:
```
[blog-creekblog] [2022-10-13 08:51:35] 10.244.14.250 - - [13/Oct/2022:08:51:35 +0000] "GET / HTTP/1.1" 200 13374 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /css/tailwind.css?id=c59db0377232c096380de81595136db8 HTTP/1.1" 200 32049 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL29jdG9wdXMuanBn?h=570&w=1472&fit=crop&s=dc1c27533a3099a87c7d2f6eda7ae047 HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL29jdG9wdXMuanBn?h=570&w=736&fit=crop&s=7c4bfa5de8e3f597171f7534ba26cbbb HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL2lkZWEuanBn?h=570&w=736&fit=crop&s=e9724f77d3b5d6382ca0193a728ae580 HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL3BpenphLXdpZmkuanBn?h=570&w=1472&fit=crop&s=b501c6e80f59d4541820bd6b03b4fe7c HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL2lkZWEuanBn?h=570&w=1472&fit=crop&s=6fccb8e28ab6480c8fe16b75c18d12b7 HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL3BpenphLXdpZmkuanBn?h=570&w=736&fit=crop&s=184aac23c563b82927cfecc15ca4cec7 HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL2JsaW5raW5nLWNhcm90LmdpZg==?h=570&w=1472&fit=crop&s=cd6a0f20b36fee65df134985e147e3da HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:37] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL2JsaW5raW5nLWNhcm90LmdpZg==?h=570&w=736&fit=crop&s=39bc605c891b80df1a5d050f4be88c90 HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:38] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL2RvbnV0LmpwZw==?h=570&w=736&fit=crop&s=ff73fb88715fb8e9853da84ec6ea4b01 HTTP/1.1" 500 6609 "https://blog.sportshaala.com/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:105.0) Gecko/20100101 Firefox/105.0
[blog-creekblog] [2022-10-13 08:51:38] 10.244.14.250 - - [13/Oct/2022:08:51:36 +0000] "GET /img/asset/YXNzZXRzL2RvbnV0LmpwZw==?h=
```

**Reply by jasonvarga** [reply]
Not server logs. The Laravel log located in `storage/logs`.

**Comment by kimili** [comment]
I just ran into this issue with a new deployment to DO App Platform. [The tutorial](https://statamic.dev/deploying/digitalocean-app-platform) mentions adding the following dependencies to `composer.json`

```
{
    "require": {
        "ext-bcmath": "*",
        "ext-exif": "*",
        "ext-gd": "*",
        "ext-mbstring": "*",
    }
}
```

What it _doesn't_ mention, and what I eventually figured out after [poking around the DO docs](https://www.digitalocean.com/community/questions/gd-library-extension-not-available-with-this-php-installation) is that after you update `composer.json`, you *also* have to run `composer update` to update the `composer.lock` file. When you commit the updates to both those files, then app platform will properly install GD and you should get the images displaying properly.

**Reply by jasonvarga** [reply]
I've updated that docs page to mention composer updating. Thanks!
